bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -1320,15 +1326,18 @@ void DWARFRewriter::updateDWARFObjectAddressRanges(
 
     // If we are at this point we are in the CU/Skeleton CU, and
     // DW_AT_GNU_ranges_base or DW_AT_rnglists_base doesn't exist.
-    if (Unit.getVersion() <= 4)
+    if (Unit.getVersion() <= 4) {
       DIEBldr.addValue(&Die, dwarf::DW_AT_GNU_ranges_base, dwarf::DW_FORM_data4,
-                       DIEInteger(*RangesBase));
-    else if (Unit.getVersion() == 5)
+                       DIEInteger(INT_MAX));
+      LegacyRangesWritersByCU[*Unit.getDWOId()]->setDie(
```

Comment:
I wonder that since this will be multi threaded, it's better to use find. [] Can write. Although in this particular case it should be read only.